Builds/CMake/CMakeFuncs.cmake

```diff
@@ -457,6 +457,9 @@ macro(setup_build_boilerplate)
       add_definitions(-DSANITIZER=TSAN)
     endif()
   endif()
+  if (NOT WIN32)
+    add_compile_options("-pthread")
```

Comment:
It's not obvious from the FindThreads docs, but I think you can alternatively use:     

`add_compile_options(${CMAKE_THREAD_LIBS_INIT})`

Also, would it make more sense to add this to `find_pthread` macro on line 293?